lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -1583,6 +1583,51 @@ static LogicalResult verify(spirv::LoadOp loadOp) {
   return verifyMemoryAccessAttribute(loadOp);
 }
 
+//===----------------------------------------------------------------------===//
+// spv.LogicalNot
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+/// Converts `spirv::LogicalNotOp` to the given `NewOp` using the first and the
+/// second operands from the given `ParentOp`.
+template <typename NewOp, typename ParentOp>
+struct ConvertLogicalNotOp : public OpRewritePattern<spirv::LogicalNotOp> {
+  using OpRewritePattern<spirv::LogicalNotOp>::OpRewritePattern;
+
+  PatternMatchResult matchAndRewrite(spirv::LogicalNotOp logicalNotOp,
+                                     PatternRewriter &rewriter) const override {
+    auto parentOp =
+        dyn_cast_or_null<ParentOp>(logicalNotOp.operand()->getDefiningOp());
+
+    if (!parentOp) {
+      return this->matchFailure();
+    }
+
+    rewriter.replaceOpWithNewOp<NewOp>(
+        /*valuesToRemoveIfDead=*/{parentOp.result()}, logicalNotOp,
+        logicalNotOp.result()->getType(), parentOp.operand1(),
+        parentOp.operand2());
+
+    return this->matchSuccess();
+  }
+};
+} // end anonymous namespace
+
+void spirv::LogicalNotOp::getCanonicalizationPatterns(
+    OwningRewritePatternList &results, MLIRContext *context) {
+  results.insert<ConvertLogicalNotOp<spirv::INotEqualOp, spirv::IEqualOp>>(
```

Comment:
These should just be chained together:

insert<ConvertLogicalNotOp<...>,
           ConvertLogicalNotOp<...>,
            ...>(context);

Reply:
@River707 thanks! Fixed.